### PR TITLE
[1.9] Introduced Capabilities for World

### PR DESCRIPTION
**Pull request by Abastro**
This will allow compatibility between mods in World part.
(PR related with my suggestion #2877, note that server part is not included here )

### CODE REVIEW

**Comment by liach** [comment]
Why to keep both prs open?

**Comment by Abastro** [comment]
@liach So, is there no possibility for PRs on 1.9 to be merged?

**Comment by liach** [comment]
Most likely for the 1.9.4 one. You may keep it open in case that 1.9.4 become mainstream.

File: src/main/java/net/minecraftforge/world/WorldCapabilityData.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	        this.data = nbt.getCompoundTag("ForgeCaps");
+	        if(serializable != null)
+	            serializable.deserializeNBT(this.data);
+	    }
+	}
+
+    @Override
+    public void writeToNBT(NBTTagCompound nbt)
+    {
+        nbt.setInteger("dimension", this.dimension);
```
Comment (williewillus) [inline src/main/java/net/minecraftforge/world/WorldCapabilityData.java]: I'm not very versed in WorldSavedData, but is this required? I thought WorldSavedData is per-world

Comment (Abastro) [inline src/main/java/net/minecraftforge/world/WorldCapabilityData.java]: It seems that I put it in to make World as INBTSerializable, and removed it later.
I'll remove the redundancy.

**Comment by Actuarius** [comment]
@williewillus added labels [Feature]

**Comment by liach** [comment]
@Abastro You should close the other one but not this as branch 1.9.4 is deserted now.

**Comment by Abastro** [comment]
@liach This branch also had merge conflicts which needs new PR.
And when I create new PR for master, it will be closed as well.

**Pull request closed without merging**